lcdtest.py

```diff
@@ -0,0 +1,168 @@
+        print('<options> i2c options:')
+        print('')
+        print('   expander - Supported I²C port expanders are PCF8574, MCP23008 and MCP23017')
+        print('   addr     - The I²C address (in hex format) can be found with')
+        print('              `i2cdetect 1` from the i2c-tools package.')
+        print('   cols     - The number of columns on your LCD, e.g. 16')
+        print('   rows     - The number of rows on your LCD, e.g. 2')
+        print('   charmap  - Charmap can be either A00 or A02. If your display contains Japanese')
+        print('              characters, it probably uses the A00 charmap, otherwise A02.')
+        print('              default=A00')
+        print('')
+        print('   Expander specific options:')
+        print('')
+        print('   MCP23017: gpio_bank - Either A or B')
+        print('')
+        print('Options format: name=value e.g. expander=MCP23008 (Example below)')
+
+        print('%s i2c charmap expander=MCP23017 addr=0x20 cols=20 rows=4 charmap=A00 gpio_bank=A' % sys.argv[0])
+    # Options for GPIO mode
+    elif ((len(sys.argv) > 1) and (sys.argv[1] == 'gpio')):
+
+        print('<options> gpio options:')
+        print('')
+        print('   mode     - GPIO numbering mode, either GPIO.BOARD or GPIO.BCM, default=GPIO.BCM')
+        print('   cols    - The number of columns on your LCD, e.g. 16')
+        print('   rows    - The number of rows on your LCD, e.g. 2')
+        print('   rs      - RS pin number')
+        print('   rw      - RW pin number')
+        print('   e       - E pin number')
+        print('   bl      - Backlight pin number, default=None')
+        print('   data    - Data (d0-d7) gpio pin numbers (4 or 8 numbers depending')
+        print('                    on if you want 4 or 8 bit mode, separated by comma)')
+        print('                    Example: data=1,2,3,4,5,6,7,8 (for 8-bit mode)')
+        print('                             data=4,5,6,7,8 (for 4-bit mode)')
+        print('   charmap - Charmap can be either A00 or A02. If your display contains Japanese')
+        print('              characters, it probably uses the A00 charmap, otherwise A02.')
+        print('              default=A00')
+        print('')
+        print('Options format: name=value e.g. expander=MCP23008 (Example below)')
+        print('%s gpio show_charmap cols=20 mode=GPIO.BCM rows=4 rs=15 rw=18 e=16 data=21,22,23,24 bl=None charmap=A00' % sys.argv[0])
+    else:
+        print('<options> For info about options run:')
+        print('')
+        print('   %s i2c' % sys.argv[0])
+        print('   %s gpio' % sys.argv[0])
+        print('')
+
+    sys.exit(1)
+
+
+if __name__ == '__main__':
+    if len(sys.argv) < 3:
+        print_usage()
+
+    lcdmode = sys.argv[1]
+    test = sys.argv[2]
+
+    # Parse options into a dictionary
+    options = {}
+    if sys.version_info > (3, 0):
+            options = dict([arg.split(sep='=', maxsplit=1) for arg in sys.argv[3:]])
```

Comment:
Ah, I see. I think that's something that automated tools can catch, once we can drop Python 2 support.